Context/Application/Command/InternalBlockchain/CheckInternalWithdrawalOperation/CheckInternalWithdrawalOperationHandler.ts

```diff
@@ -0,0 +1,121 @@
+            await this.checkTransferOperation(transferOperation, command.withdraw);
+
+            command.withdraw.readyToProcess(
+                htlcOperation.get("op")[1].preimage_hash[1],
+                htlcOperation.get("op")[1].claim_period_seconds,
+                htlcOperation.get("op")[1].amount.amount,
+                transferOperation.get("op")[1].amount.amount,
+                transferOperation.get("op")[1].amount.asset_id
+            );
+        } catch (error: unknown) {
+            if (error instanceof Error) {
+                command.withdraw.error(error.message);
+            }
+        }
+
+        this.withdrawRepository.save(command.withdraw);
+    }
+
+    public async checkHTLCOperation(htlcOperation: Map<string, any>, withdraw: Withdraw) {
+        if (!htlcOperation) {
+            throw new Errors.HTLCCreateOperationNotFound(withdraw);
+        }
+
+        await this.checkLastIrreversible(htlcOperation);
+        await this.checkReceiver(htlcOperation);
+
+        const normalizedAmount = this.normalizer.normalize(
+            htlcOperation.get("op")[1].amount.amount,
+            await this.internalBlockchain.getAsset(htlcOperation.get("op")[1].amount.asset_id)
+        );
+
+        if (normalizedAmount < config.eth.minimum_withdraw_amount) {
+            throw new Errors.InvalidAmount(withdraw);
+        }
+
+        if (htlcOperation.get("op")[1].amount.asset_id != "1.3.1") {
+            throw new Errors.InvalidAsset(withdraw);
+        }
+
+        if (htlcOperation.get("op")[1].claim_period_seconds < config.contract.withdrawn_timelock) {
+            throw new Errors.InvalidTimelock(withdraw);
+        }
+
+        if (!htlcOperation.get("op")[1].preimage_hash[1]) {
+            throw new Errors.InvalidHashlock(withdraw);
+        }
+
+        if (htlcOperation.get("op")[1].preimage) {
+            throw new Errors.InvalidPreimage(withdraw);
+        }
+    }
+
+    private async checkTransferOperation(transferOperation: Map<string, any>, withdraw: Withdraw) {
+        await this.checkLastIrreversible(transferOperation);
+        await this.checkReceiver(transferOperation);
+
+        let minimalWithdrawalFee = config.revpop.rveth_withdrawal_fee;
+        if (transferOperation.get("op")[1].amount.asset_id == "1.3.0") {
+            minimalWithdrawalFee = config.revpop.rvp_withdrawal_fee;
+        }
+
+        if (transferOperation.get("op")[1].amount.amount < minimalWithdrawalFee) {
+            throw new Errors.InvalidWithdrawalFee(withdraw);
+        }
+    }
+
+    private async checkLastIrreversible(operation: Map<string, any>) {
+        if (!this.lastIrreversibleBlockNumber) {
+            this.lastIrreversibleBlockNumber = await this.internalBlockchain.getLastIrreversibleBlockNumber();
+        }
+
+        if (operation.get("block_num") > this.lastIrreversibleBlockNumber) {
+            throw new Errors.BlockIsReversible(operation.get("id"));
```

Comment:
Do not hard fail here. Need to check the contract later.

Reply:
It is not hard fail, contracts will rechecked on next run